sotodlib/toast/workflows/proc_mapmaker.py

```diff
@@ -191,103 +193,126 @@ def mapmaker_run(job, otherargs, runargs, data, map_op):
+                    comm=data.comm.comm_world,
                 )
-                # Data object that only covers one observation
-                obs_data = data.select(obs_uid=obs.uid)
-                # Replace comm_world with the group communicator
-                obs_data._comm = new_comm
-                binner = map_op.binning
-                orig_view = binner.pixel_pointing.view
-                if do_intervalmaps and orig_view is not None:
-                    if isinstance(map_op, so_ops.Splits):
-                        msg = "Interval mapping cannot be used with Splits"
-                        raise RuntimeError(msg)
-                    # Map each interval separately
-                    ob = obs_data.obs[0]
-                    times = ob.shared[defaults.times].data
-                    views = ob.intervals[orig_view]
-                    for iview, view in enumerate(views):
-                        # Add a view for this specific interval
-                        single_view = f"{orig_view}-{iview}"
-                        ob.intervals[single_view] = toast.IntervalList(
-                            times, timespans=[(view.start, view.stop)]
-                        )
-                        binner.pixel_pointing.view = single_view
-                        map_op.name = f"{orig_name}_{obs.name}-{iview}"
-                        map_op.reset_pix_dist = True
-                        try:
-                            map_op.apply(obs_data)
-                            log.info_rank(
-                                f"{group} : Mapped "
-                                f"{obs.name}-{iview} / {len(views)} in",
-                                comm=new_comm.comm_world,
-                                timer=timer_obs,
-                            )
-                        except Exception as e:
-                            log.info_rank(
-                                f"{group} : Failed to map "
-                                f"{obs.name}-{iview} / {len(views)} (e) in",
-                                comm=new_comm.comm_world,
-                                timer=timer_obs,
+                # Map each observation separately
+                timer_obs = toast.timing.Timer()
+                timer_obs.start()
+                group = data.comm.group
+                orig_name = map_op.name
+                orig_comm = data.comm
+                new_comm = toast.Comm(world=data.comm.comm_group)
+                for iobs, obs in enumerate(data.obs):
+                    log.info_rank(
+                        f"{group} : mapping observation {iobs + 1} "
+                        f"/ {len(data.obs)}.",
+                        comm=new_comm.comm_world,
+                    )
+                    # Data object that only covers one observation
+                    obs_data = data.select(obs_uid=obs.uid)
+                    # Replace comm_world with the group communicator
+                    obs_data._comm = new_comm
```

Comment:
I know this was in the original code, but I think that this only works "accidentally" and is fragile (there is a reason this is an internal variable).  The `Data` constructor assigns the internal `_comm` variable to the input `Comm` object.  However, each observation has a reference to this original `Comm` instance as well.  And every Observation shared object has references to the original sub-communicators.  If we want to support getting a subset of observations on each group where we pretend that the group communicator is the world communicator, then we should probably make that kind of change within the `Data` class where we can do that more carefully and in one place.  Again, this is no worse than the original code, so it is fine to merge, but we should implement something cleaner upstream.